src/lib/y2storage/filesystems/btrfs.rb

```diff
@@ -80,6 +80,36 @@ def supports_btrfs_subvolumes?
         true
       end
 
+      # Check if a subvolume can be created.
+      #
+      # It can always be created if we're going to create the whole file
+      # system anyway.
+      #
+      # If the file system exists already there must at least be nothing
+      # else below path.
+      #
+      # @param path [String] subvolume path
+      #
+      # @return [Boolean]
+      #
+      def subvolume_can_be_created?(path)
+        return true unless exists_in_probed?
```

Comment:
I prefer to restrict the usage of `#exists_in_probed?` as much as possible. In general we would like to have code that can be used to calculate stuff taking any devicegraph as starting point, not only TheOneAndOnlyProbed . It's more testable and flexible.

For example, [BlkFilesystem#to_be_formatted?](https://github.com/yast/yast-storage-ng/blob/21a499e61698ffcf4d73d1ad532e501062371dea/src/lib/y2storage/blk_device.rb#L396) gets the initial devicegraph as argument. In the Partitioner it will get `Y2Partitioner::DeviceGraphs.system`. In the proposal that would be whatever was the starting point for the proposal (which strictly speaking doesn't HAVE to be probed). In a test it can be another thing.

Reply:
`#create_btrfs_subvolume` doesn't know a devicegraph. So `in_probed?` is the best to do, and the semantics is correct. It compares to the situation the devicegraph commit will encounter later.